docs/architecture/adr-002-evm-hooks.md

```diff
@@ -0,0 +1,52 @@
+# ADR 002: EVM Hooks
+
+## Changelog
+
+- 2021-08-11: first draft
+
+## Status
+
+PROPOSED
+
+## Abstract
+
+The current ADR proposes a hook interface to EVM module, with the goal of extend the functionality externally,
+specifically to support communication from EVM contract to native modules through logs.
+
+## Context
+
+There are some requirements to enable EVM contract to communicate with cosmos native modules, which could have multiple
+use cases. We need some kind of hooks to support these extension requirements in a generic way.
```

Comment:
explain use cases

Reply:
Added more detail to the "Call Native Module" use case.